app/services/wallets/validate_service.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Wallets
+  class ValidateService
+    def initialize(result, **args)
+      @result = result
+      @args = args
+    end
+
+    def valid?
+      errors = []
+      errors << valid_customer?
+      errors << valid_paid_credits_amount?
+      errors << valid_granted_credits_amount?
+      errors = errors.compact
+
+      unless errors.empty?
+        result.fail!(
+          code: 'unprocessable_entity',
+          message: 'Validation error on the record',
+          details: errors,
+        )
+        return false
+      end
+
+      true
+    end
+
+    private
+
+    attr_accessor :result, :args
+
+    def valid_customer?
+      current_customer = Customer.find_by(
+        id: args[:customer_id],
+        organization_id: args[:organization_id],
+      )
+
+      return 'customer_not_found' unless current_customer
+      return 'wallet_already_exists' if current_customer.wallets.active.any?
+      return 'no_active_subscription' unless current_customer.subscriptions.active.any?
```

Comment:
I think we can use `exists?` here instead of `any?`, so that it works at a SQL level and not use ruby enumerable